src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/ElementOfMutatorFactory.java

```diff
@@ -0,0 +1,143 @@
+import java.lang.reflect.AnnotatedType;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+final class ElementOfMutatorFactory implements MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(
+      AnnotatedType type, ExtendedMutatorFactory factory) {
+    ElementOf elementOf = type.getAnnotation(ElementOf.class);
+    if (elementOf == null) {
+      return Optional.empty();
+    }
+    if (!(type.getType() instanceof Class<?>)) {
+      return Optional.empty();
+    }
+    Class<?> rawType = (Class<?>) type.getType();
+
+    if (rawType == byte.class || rawType == Byte.class) {
+      return Optional.of(
+          elementOfMutator(boxBytes(elementOf.bytes()), "bytes", rawType.getSimpleName()));
+    } else if (rawType == short.class || rawType == Short.class) {
+      return Optional.of(
+          elementOfMutator(boxShorts(elementOf.shorts()), "shorts", rawType.getSimpleName()));
+    } else if (rawType == int.class || rawType == Integer.class) {
+      return Optional.of(
+          elementOfMutator(boxInts(elementOf.integers()), "integers", rawType.getSimpleName()));
+    } else if (rawType == long.class || rawType == Long.class) {
+      return Optional.of(
+          elementOfMutator(boxLongs(elementOf.longs()), "longs", rawType.getSimpleName()));
+    } else if (rawType == char.class || rawType == Character.class) {
+      return Optional.of(
+          elementOfMutator(boxChars(elementOf.chars()), "chars", rawType.getSimpleName()));
+    } else if (rawType == float.class || rawType == Float.class) {
+      return Optional.of(
+          elementOfMutator(boxFloats(elementOf.floats()), "floats", rawType.getSimpleName()));
+    } else if (rawType == double.class || rawType == Double.class) {
+      return Optional.of(
+          elementOfMutator(boxDoubles(elementOf.doubles()), "doubles", rawType.getSimpleName()));
+    } else if (rawType == String.class) {
+      return Optional.of(
+          elementOfMutator(Arrays.asList(elementOf.strings()), "strings", rawType.getSimpleName()));
+    }
+    return Optional.empty();
+  }
+
+  private static <T> SerializingMutator<T> elementOfMutator(
+      List<T> values, String fieldName, String targetTypeName) {
+    require(
+        values.size() > 1,
+        format(
+            "@ElementOf %s array must contain at least two values for %s",
+            fieldName, targetTypeName));
+
+    return mutateThenMap(
+        mutateIndices(values.size()),
+        values::get,
+        value -> {
+          int index = values.indexOf(value);
+          require(
+              index >= 0,
+              "@ElementOf produced value not contained in the declared value set for "
+                  + targetTypeName);
+          return index;
+        },
+        isInCycle ->
+            format("@ElementOf(%s, size=%d) -> %s", fieldName, values.size(), targetTypeName));
```

Comment:
`@ElementOf(strings, size=10) -> String`
WDYT of removing redundant part "strings, "?:
`@ElementOf(size=10) -> String`

Or even something like this: `@ElementOf<String>(10)`

Reply:
Your tight, this can be tightened up. The last one is still very readable. :+1: